lib/bigtable/family.js

```diff
@@ -0,0 +1,257 @@
+
+  var methods = {
+
+    /**
+     * Create a column family.
+     *
+     * @param {object=} options - See {module:bigtable/table#createFamily}.
+     *
+     * @example
+     * family.create(function(err, family, apiResponse) {
+     *   // The column family was created successfully.
+     * });
+     */
+    create: true,
+
+    /**
+     * Delete the column family.
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * family.delete(function(err, apiResponse));
+     */
+    delete: {
+      protoOpts: {
+        service: 'BigtableTableService',
+        method: 'deleteColumnFamily'
+      },
+      reqOpts: {
+        name: this.name
+      }
+    },
+
+    /**
+     * Check if the column family exists.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {boolean} callback.exists - Whether the family exists or not.
+     *
+     * @example
+     * family.exists(function(err, exists) {});
+     */
+    exists: true,
+
+    /**
+     * Get a column family if it exists.
+     *
+     * You may optionally use this to "get or create" an object by providing an
+     * object with `autoCreate` set to `true`. Any extra configuration that is
+     * normally required for the `create` method must be contained within this
+     * object as well.
+     *
+     * @param {options=} options - Configuration object.
+     * @param {boolean} options.autoCreate - Automatically create the object if
+     *     it does not exist. Default: `false`
+     *
+     * @example
+     * family.get(function(err, family, apiResponse) {
+     *   // The `family` data has been updated.
+     * });
+     */
+    get: true
+  };
+
+  var config = {
+    parent: table,
+    id: this.name,
+    methods: methods,
+    createMethod: function(_, callback) {
+      table.createFamily(name, callback);
+    }
+  };
+
+  GrpcServiceObject.call(this, config);
+}
+
+nodeutil.inherits(Family, GrpcServiceObject);
+
+/**
+ * @private
+ */
+Family.formatName_ = function(table, name) {
+  if (name.indexOf('/') > -1) {
+    return name;
+  }
+
+  return table.name + '/columnFamilies/' + name;
+};
+
+/**
+ * @private
+ */
+Family.parseRule_ = function(ruleObj) {
+  var rules = [];
+
+  if (ruleObj.age) {
+    rules.push(createRule('maxAge', ruleObj.age));
+  }
+
+  if (ruleObj.versions) {
+    rules.push(createRule('maxNumVersions', ruleObj.versions));
+  }
+
+  if (!ruleObj.intersection && !ruleObj.union) {
+    return rules[0].rule;
+  }
+
+  var rule = {};
+  var ruleType = ruleObj.union ? 'union' : 'intersection';
+
+  rule[ruleType] = {
+    rules: rules
+  };
+
+  return rule;
+
+  function createRule(key, value) {
```

Comment:
Is it cool if we just write the object this creates out manually where `createRule` is being used?